File: No.2490-Bees-master/api/src/main/java/org/bees/biz/service/OrderLogService.java
package org.bees.biz.service;

import java.lang.reflect.InvocationTargetException;
import java.util.Date;
import java.util.LinkedList;
import java.util.List;

import org.apache.commons.beanutils.PropertyUtils;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.bees.biz.persistence.manager.OrderLogManager;
import org.bees.biz.persistence.manager.UserManager;
import org.bees.biz.persistence.model.OrderLog;
import org.bees.biz.persistence.model.User;
import org.bees.biz.service.dto.OrderLogDto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import projects.commons.utils.ValidateUtils;

@Service
public class OrderLogService {
	
	@Autowired
	private OrderLogManager orderLogManager;
	
	@Autowired
	private UserManager userManager;
	
	@SuppressWarnings("unused")
	private final static Log LOG = LogFactory.getLog(OrderLogService.class);

	public OrderLog getInfoById(Integer id) {		
		return this.orderLogManager.getInfoById(id);
	}

	public List<OrderLogDto> getList(Integer uid, Byte type, Byte incomeType,
			String mobile, String nickname, Date startDate, Date endDate,
			String startMoney, String endMoney, String orderId, Integer start,
			Integer limit) {
		List<User> users=this.userManager.adminList(mobile, nickname,null, null, null, null, null,null, uid, null,null,null, null);
		List<Integer> uids=null;
		if(!ValidateUtils.isNull(users)){
			uids=new LinkedList<Integer>();
			for (User user : users) {
				uids.add(user.getId());
			}
		}
		List<OrderLog> OrderLogs=this.orderLogManager.getList(uids,type,incomeType,startDate,endDate,startMoney,endMoney,orderId,start,limit);
		List<OrderLogDto> dtos=new LinkedList<OrderLogDto>();

		for (OrderLog orderLog : OrderLogs) {
			OrderLogDto dto=new OrderLogDto();
			User user=this.userManager.getInfoById(orderLog.getUid());
			try {
				PropertyUtils.copyProperties(dto, user);
				PropertyUtils.copyProperties(dto, orderLog);
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}		
		return dtos;
	}

	public boolean del(Integer id) {		
		return this.orderLogManager.del(id);
	}

	public int getListCount(Integer uid, Byte type, Byte incomeType,
			String mobile, String nickname, Date startDate, Date endDate,
			String startMoney, String endMoney, String orderId) {
		List<User> users=this.userManager.adminList(mobile, nickname,null, null, null, null, null,null, uid, null,null,null, null);
		List<Integer> uids=null;
		if(!ValidateUtils.isNull(users)){
			uids=new LinkedList<Integer>();
			for (User user : users) {
				uids.add(user.getId());
			}
		}
		return this.orderLogManager.getListCount(uids,type,incomeType,startDate,endDate,startMoney,endMoney,orderId);
	}


	
	
}

File: beesInterface/bees_orderLog/orderLogService.py
#coding:utf-8
from __future__ import unicode_literals
from models import OrderLog
from utils.validateUtils import ValidateUtils
from bees_user.models import User
from utils.const import Const

class OrderLogService(object):

	@staticmethod
	def getInfoById(id):
		try:
			return OrderLog.objects.get(id=id, del_flag=Const.DEL_FLAG_NOT_DEL)
		except:
			return None

	@staticmethod
	def getList(uid, type, incomeType, mobile,nickname, startDate, endDate, startMoney, endMoney, orderId, start, limit):
		users = User.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(mobile):
			users = users.filter(mobile=mobile)
		if not ValidateUtils.isNull(nickname):
			users = users.filter(nick_name__contains=nickname)
		if not ValidateUtils.isNull(uid):
			users = users.filter(id=int(uid))
		uids = []
		if not ValidateUtils.isNull(users):
			for user in users:
				uids.append(user.id)
		orderLogs = OrderLog.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(uids):
			orderLogs = orderLogs.filter(uid__in=uids)
		if not ValidateUtils.isNull(type):
			orderLogs = orderLogs.filter(type=type)
		if not ValidateUtils.isNull(incomeType):
			orderLogs = orderLogs.filter(income_type=incomeType)
		if not ValidateUtils.isNull(startDate):
			orderLogs = orderLogs.filter(add_time__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			orderLogs = orderLogs.filter(add_time__lte=endDate)
		if not ValidateUtils.isNull(startMoney):
			orderLogs = orderLogs.filter(money__gte=startMoney)
		if not ValidateUtils.isNull(endMoney):
			orderLogs = orderLogs.filter(money__lte=endMoney)
		if not ValidateUtils.isNull(orderId):
			orderLogs = orderLogs.filter(order_id=orderId)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			orderLogs = orderLogs[start:]
		else:
			limit = int(limit)
			orderLogs = orderLogs[start: start+limit]
		dtos=[]
		for orderLog in orderLogs:
			dtos.append(orderLog.getOrderLogDto())
		return dtos

	@staticmethod
	def delete(id):
		try:
			orderlog = OrderLog.objects.get(id=id)
			orderlog.delete()
		except:
			return False
		return True

	@staticmethod
	def getListCount(uid, type, incomeType, mobile, nickname, startDate, endDate, startMoney, endMoney, orderId):
		users = User.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(mobile):
			users = users.filter(mobile=mobile)
		if not ValidateUtils.isNull(nickname):
			users = users.filter(nick_name__contains=nickname)
		if not ValidateUtils.isNull(uid):
			users = users.filter(id=int(uid))
		uids = []
		if not ValidateUtils.isNull(users):
			for user in users:
				uids.append(user.id)
		orderLogs = OrderLog.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL)
		if not ValidateUtils.isNull(uids):
			orderLogs = orderLogs.filter(uid__in=uids)
		if not ValidateUtils.isNull(type):
			orderLogs = orderLogs.filter(type=type)
		if not ValidateUtils.isNull(incomeType):
			orderLogs = orderLogs.filter(income_type=incomeType)
		if not ValidateUtils.isNull(startDate):
			orderLogs = orderLogs.filter(add_time__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			orderLogs = orderLogs.filter(add_time__lte=endDate)
		if not ValidateUtils.isNull(startMoney):
			orderLogs = orderLogs.filter(money__gte=startMoney)
		if not ValidateUtils.isNull(endMoney):
			orderLogs = orderLogs.filter(money__lte=endMoney)
		if not ValidateUtils.isNull(orderId):
			orderLogs = orderLogs.filter(order_id=orderId)
		return orderLogs.count()